ukvm/ukvm_hv_kvm_x86_64.c

```diff
@@ -194,12 +194,20 @@ void ukvm_hv_vcpu_loop(struct ukvm_hv *hv)
                 errx(1, "Invalid guest port access: port=0x%x", run->io.port);
 
             int nr = run->io.port - UKVM_HYPERCALL_PIO_BASE;
+            ukvm_gpa_t gpa =
+                *(uint32_t *)((uint8_t *)run + run->io.data_offset);
+
+            /* Guest has halted the CPU. */
+            if (nr == UKVM_HYPERCALL_HALT) {
```

Comment:
Please do the declaration and load of `gpa` at the start of this `if` block, and keep the original declaration and load of `gpa` below, before the call to `fn(...)`.

Reply:
changed